src/getTokenList.ts

```diff
@@ -1,33 +1,42 @@
-import { TokenList } from "@uniswap/token-lists";
-import fs from "fs";
-import hre from "hardhat";
-
-import { TrancheFactory__factory } from "elf-contracts-typechain/dist/types/factories/TrancheFactory__factory";
-
-import { AddressesJsonFile } from "src/addresses/AddressesJsonFile";
-
-import { getYieldTokenInfos } from "./yieldTokens";
-import { getPrincipalTokenInfos } from "./principalTokens";
+import { TokenList } from "@uniswap/token-lists/src";
 import {
   ConvergentPoolFactory__factory,
   Vault__factory,
   WeightedPoolFactory__factory,
 } from "elf-contracts-typechain/dist/types";
+import { TrancheFactory__factory } from "elf-contracts-typechain/dist/types/factories/TrancheFactory__factory";
+import fs from "fs";
+import hre from "hardhat";
+import { AddressesJsonFile } from "src/addresses/AddressesJsonFile";
 import { getAssetProxyTokenInfos } from "src/assetProxies";
 import { getPrincipalPoolTokenInfos } from "src/ccpools";
-import { getUnderlyingTokenInfos } from "src/underlying";
+import { ELEMENT_LOGO_URI } from "src/logo";
+import { TokenTag } from "src/tags";
+import { BaseTokenInfo, CurveBaseToken, TagInfo } from "src/types";
 import { getVaultTokenInfos } from "src/vaults";
 import { getYieldPoolTokenInfos } from "src/weightedPools";
-import { TokenTag } from "src/tags";
-import { TagInfo } from "src/types";
-import { ELEMENT_LOGO_URI } from "src/logo";
+import { getBaseTokenInfos } from "./baseTokens";
+import { getPrincipalTokenInfos } from "./principalTokens";
+import { getYieldTokenInfos } from "./yieldTokens";
+
+import { ethers } from "ethers";
+import { getRootTokenInfos } from "./rootTokens";
 
 const provider = hre.ethers.provider;
 
 export const elementTags: Record<TokenTag, TagInfo> = {
-  [TokenTag.UNDERLYING]: {
-    name: "Underlying asset",
-    description: "The base asset of a principal or yield token",
+  [TokenTag.BASE]: {
+    name: "Base token",
+    description: "The underlying base asset of a principal or yield token",
+  },
+  [TokenTag.ROOT]: {
+    name: "Root token",
+    description:
+      "This token is a member of a given LP token's pool assets. The LP token for that pool must also be a base token",
+  },
+  [TokenTag.CURVE]: {
+    name: "Curve LP token",
+    description: "A curve LP token",
```

Comment:
Ohh I really like the `CURVE` tag for base assets that are LP tokens. That's a better model!

And I think we can get rid of `BASE` and `ROOT` tags entirely now. For tokens like dai, usdc, lusd, etc.. we don't need any tags or extensions on them. This basically treats them as boring erc20s, which is exactly what they are and token infos for `eP`, `eY`, `CURVE`, `CCPOOL`, etc.. will just reference them in their extensions.

Reply:
Hmm, I think we actually do need them, specifically in the case of `3crv` cos it's an LP token that is a root token, as in dependent on a base token like `MIM-3LP`